### ISSUE TITLE
Intermittent timeout in /websockets/constructor/010.html

### ISSUE THREAD

**Issue by jdm** [servo/servo#9902]
```
  ▶ TIMEOUT [expected OK] /websockets/constructor/010.html

  ▶ Unexpected subtest result in /websockets/constructor/010.html:
  │ TIMEOUT [expected FAIL] WebSockets: protocol in response but no requested protocol
  └   → Test timed out
```

**Comment by perlun** [comment]
Is this the same underlying issue as #9777? Sounds fairly likely I guess?

**Comment by jdm** [comment]
Yes, I would expect all of the intermittent websocket timeouts to be symptoms of #9777.

**Comment by perlun** [comment]
FWIW: I did a bit of playing around in https://github.com/bobthekingofegypt/servo-cookie-inter/pull/1.

**Comment by jdm** [comment]
Should be fixed by #10240.
